### PR TITLE
[Admin][Product] Enable tracking

### PR DESCRIPTION
**Pull request by Arminek**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Related tickets | -
| License         | MIT

### CODE REVIEW

File: features/product/managing_product_variants/disable_or_enable_inventory_on_product_variant.feature [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    I want to be able to disable or enable inventory tracking
+
+    Background:
+        Given the store is available in "English (United States)"
+        And the store has a "Wyborowa Vodka" configurable product
+        And the product "Wyborowa Vodka" has "Wyborowa Vodka Exquisite" variant priced at "$40.00"
+        And I am logged in as an administrator
+
+    @ui
+    Scenario: Disabling inventory for product variant
```
Comment (tuka217) [inline features/product/managing_product_variants/disable_or_enable_inventory_on_product_variant.feature]: (...) for **a** product (...) ?

File: features/product/managing_product_variants/disable_or_enable_inventory_on_product_variant.feature [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        And I want to modify the "Wyborowa Vodka Exquisite" product variant
+        When I disable its tracking
+        And I save my changes
+        Then I should be notified that it has been successfully edited
+        And this variant should not be tracked
+
+    @ui
+    Scenario: Enabling inventory for product variant
+        Given I want to modify the "Wyborowa Vodka Exquisite" product variant
+        When I enable its tracking
```
Comment (pjedrzejewski) [inline features/product/managing_product_variants/disable_or_enable_inventory_on_product_variant.feature]: inventory tracking

File: features/product/managing_product_variants/disable_or_enable_inventory_on_product_variant.feature [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        And I save my changes
+        Then I should be notified that it has been successfully edited
+        And this variant should be tracked
```
Comment (pjedrzejewski) [inline features/product/managing_product_variants/disable_or_enable_inventory_on_product_variant.feature]: And inventory of this variant should be tracked

File: features/product/managing_product_variants/disable_or_enable_inventory_on_product_variant.feature [hunk 4]
```diff
@@ -0,0 +1,28 @@
+@managing_product_variants
+Feature: Enable the inventory tracking
+    In order to enable inventory tracking
+    As an Administrator
+    I want to be able to disable or enable inventory tracking
+
+    Background:
+        Given the store is available in "English (United States)"
+        And the store has a "Wyborowa Vodka" configurable product
+        And the product "Wyborowa Vodka" has "Wyborowa Vodka Exquisite" variant priced at "$40.00"
```
Comment (lchrusciel) [inline features/product/managing_product_variants/disable_or_enable_inventory_on_product_variant.feature]: 👍 For product samples

File: features/product/managing_product_variants/disable_or_enable_inventory_on_product_variant.feature [hunk 5]
```diff
@@ -0,0 +1,28 @@
+@managing_product_variants
+Feature: Enable the inventory tracking
+    In order to enable inventory tracking
+    As an Administrator
+    I want to be able to disable or enable inventory tracking
```
Comment (lchrusciel) [inline features/product/managing_product_variants/disable_or_enable_inventory_on_product_variant.feature]: I want to be able * (1 + (to dis) + (or en)) inventory tracking

Comment (lchrusciel) [inline features/product/managing_product_variants/disable_or_enable_inventory_on_product_variant.feature]: `I want to be enable inventory tracking`?

Comment (pamil) [inline features/product/managing_product_variants/disable_or_enable_inventory_on_product_variant.feature]: `(?:dis|en)able`

Comment (lchrusciel) [inline features/product/managing_product_variants/disable_or_enable_inventory_on_product_variant.feature]: Oh, ok. The feature title here is `Enable the inventory tracking` but feature file name suggest both actions `disable_or_enable_inventory_on_product_variant`.

Comment (lchrusciel) [inline features/product/managing_product_variants/disable_or_enable_inventory_on_product_variant.feature]:
```gherkin
    In order to change an inventory tracking state
    As an Administrator
    I want to toggle inventory tracking
```
? /cc @TheMadeleine

File: src/Sylius/Behat/Context/Setup/ProductContext.php [hunk 6]
```diff
@@ -413,7 +413,28 @@ public function thereAreItemsOfProductInVariantAvailableInTheInventory($quantity
...
         $this->objectManager->flush();
     }
-    
+
+    /**
+     * @Given /^the ("[^"]+" product) is tracked by the inventory$/
+     */
+    public function theProductIsTrackedByTheInventory(ProductInterface $product)
+    {
+        $productVariant = $product->getFirstVariant();
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Setup/ProductContext.php]: What do you think about using a variant resolver here?

File: features/product/managing_product_variants/disable_or_enable_inventory_on_product_variant.feature (hunk 5 above)
Comment (TheMadeleine) [inline features/product/managing_product_variants/disable_or_enable_inventory_on_product_variant.feature]:
```
In order to have the inventory tracked in my shop
As an Administrator
I want to toggle the inventory tracking
```
?

File: src/Sylius/Bundle/AdminBundle/Resources/views/ProductVariant/Field/inventory.html.twig [hunk 7]
```diff
@@ -7,3 +8,9 @@
     </div>
     {% endif %}
 </div>
+{% else %}
+    <span class="ui red label">
+        <i class="remove icon"></i>
+        {{ 'sylius.ui.untraceable'|trans }}
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AdminBundle/Resources/views/ProductVariant/Field/inventory.html.twig]: Do you have translation for it?

File: features/product/managing_product_variants/disable_or_enable_inventory_on_product_variant.feature (hunk 4 above)
Comment (TheMadeleine) [inline features/product/managing_product_variants/disable_or_enable_inventory_on_product_variant.feature]: ``has a "Wyborowa Vodka Exquisite" variant priced at "$40.00"``

File: src/Sylius/Behat/Context/Ui/Admin/ManagingProductVariantsContext.php [hunk 8]
```diff
@@ -213,12 +229,12 @@ public function productShouldExistInTheProductCatalog(ProductVariantInterface $p
     }
 
     /**
-     * @When I want to modify the :productVariant product variant
+     * @When /^I want to modify the ("[^"]+" product variant)$/
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingProductVariantsContext.php]: Why did you change it? Didn't it work?

File: features/product/managing_products/disable_or_enable_inventory_on_product.feature [hunk 9]
```diff
@@ -0,0 +1,27 @@
+@managing_products
+Feature: Enable the inventory tracking
```
Comment (TheMadeleine) [inline features/product/managing_products/disable_or_enable_inventory_on_product.feature]: Enabling?

File: features/product/managing_product_variants/disable_or_enable_inventory_on_product_variant.feature (hunk 3 above)
Comment (TheMadeleine) [inline features/product/managing_product_variants/disable_or_enable_inventory_on_product_variant.feature]: *the inventory

File: features/product/managing_products/disable_or_enable_inventory_on_product.feature [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    I want to be able to disable or enable inventory tracking
+
+    Background:
+        Given the store is available in "English (United States)"
+        And the store has a product "Dice Brewing"
+        And I am logged in as an administrator
+
+    @ui
+    Scenario: Disabling inventory for simple product
+        Given the "Dice Brewing" product is tracked by the inventory
```
Comment (TheMadeleine) [inline features/product/managing_products/disable_or_enable_inventory_on_product.feature]: ``Given the inventory of the "Dice Brewing" product is tracked`` ?

File: src/Sylius/Behat/Context/Ui/Admin/ManagingProductVariantsContext.php [hunk 11]
```diff
@@ -319,6 +319,28 @@ public function thisProductVariantShouldBeTracked(ProductVariantInterface $produ
     }
 
     /**
+     * @Then /^I should see that the ("([^"]+)" variant) is not tracked$/
```
Comment (pamil) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingProductVariantsContext.php]: `("([^"]+)" variant)` -> `("[^"]+" variant)`

File: src/Sylius/Behat/Context/Ui/Admin/ManagingProductVariantsContext.php [hunk 12]
```diff
@@ -319,6 +319,28 @@ public function thisProductVariantShouldBeTracked(ProductVariantInterface $produ
...
+    public function iShouldSeeThatIsNotTracked(ProductVariantInterface $productVariant)
+    {
+        Assert::true(
+            $this->indexPage->isSingleResourceOnPage(['name' => $productVariant->getName(), 'inventory' => 'Not tracked']),
+            sprintf('This "%s" variant should have label not tracked, but it does not have', $productVariant->getName())
+        );
+    }
+
+    /**
+     * @Then /^I should see that the ("[^"]+" variant) has zero on hand quantity$/
```
Comment (pamil) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingProductVariantsContext.php]: `("([^"]+)" variant)` -> `("[^"]+" variant)`

**Comment by pjedrzejewski** [comment]
Thank you Arek! :)
